### ISSUE TITLE
dn_pos_idx calculattion for loss

### ISSUE THREAD

**Issue by jsdtwxy** [IDEACVR/DINO#48]
for i in range(len(targets)):
                if len(targets[i]['labels']) > 0:
                    t = torch.arange(0, len(targets[i]['labels']) - 1).long().cuda()
                    t = t.unsqueeze(0).repeat(scalar, 1)
                    tgt_idx = t.flatten()
                    output_idx = (torch.tensor(range(scalar)) * single_pad).long().cuda().unsqueeze(1) + t
                    output_idx = output_idx.flatten()
                else:
                    output_idx = tgt_idx = torch.tensor([]).long().cuda()

This is the code. I can't understand the subtraction on the third line. Why is the target number reduced by 1.

**Comment by SuperHenry2333** [comment]
torch.arange(a,b) generate a list [a, a+1, ..., b-1, b] including b, which is different from range(a,b).

**Comment by jsdtwxy** [comment]
> torch.arange(a,b) generate a list [a, a+1, ..., b-1, b] including b, which is different from range(a,b).

As far as I know, just the opposite is true.The method 'arange' includes the end point while the method 'range' not.
torch.arange(a,b) =  [a, a+1, ..., b-1], torch.range(a,b) =  [a, a+1, ..., b-1, b]

**Comment by SuperHenry2333** [comment]
> for i in range(len(targets)):

Sorry, my typo. Yes torch.range(a,b) = [a, a+1, ..., b-1, b]. I think we use torch.range in our code. But in the code segment you display, you use torch.arange.
